build.cake

```diff
@@ -0,0 +1,408 @@
+
+var target = Argument("target", "Default");
+var configuration = Argument("configuration", "Release");
+var testConfiguration = Argument("test-configuration", "Debug");
+
+var environment = new CakeEnvironment();
+
+var shell = IsRunningOnWindows() ? "powershell" : "bash";
+var shellArgument = IsRunningOnWindows() ? "/Command" : "-C";
+var shellExtension = IsRunningOnWindows() ? "ps1" : "sh";
+
+public class BuildPlan
+{
+    public IDictionary<string, string[]> TestProjects { get; set; }
+    public string BuildToolsFolder { get; set; }
+    public string ArtifactsFolder { get; set; }
+    public string DotNetFolder { get; set; }
+    public string[] Frameworks { get; set; }
+    public string[] Rids { get; set; }
+    public string MainProject { get; set; }
+}
+
+var buildPlan = DeserializeJsonFromFile<BuildPlan>($"{environment.WorkingDirectory}/build.json");
+
+var dotnetFolder = $"{environment.WorkingDirectory}/{buildPlan.DotNetFolder}";
+var dotnetcli = IsRunningOnWindows() ? $"{dotnetFolder}/cli/bin/dotnet" :
+                    $"{dotnetFolder}/bin/dotnet";
+var toolsFolder = $"{environment.WorkingDirectory}/tools";
+var xunitRunner = "xunit.runner.console";
+
+var sourceFolder = $"{environment.WorkingDirectory}/src";
+var testFolder = $"{environment.WorkingDirectory}/tests";
+
+var artifactFolder = $"{environment.WorkingDirectory}/{buildPlan.ArtifactsFolder}";
+var publishFolder = $"{artifactFolder}/publish";
+var logFolder = $"{artifactFolder}/logs";
+var packageFolder = $"{artifactFolder}/package";
+var installFolder = IsRunningOnWindows() ? $"{environment.WorkingDirectory}/fakeinstall/local" : "~/.omnisharp/local";
+
+string GetLocalRuntimeID(string dotnetcli)
+{
+    var process = StartAndReturnProcess(dotnetcli, 
+        new ProcessSettings
+        { 
+            Arguments = "--version",
+            RedirectStandardOutput = true
+        });
+    process.WaitForExit();
+    if (process.GetExitCode() != 0)
+        throw new Exception("Failed to get run dotnet --version");
+    foreach (var line in process.GetStandardOutput())
+    {
+        if (!line.Contains("Runtime Id"))
+            continue;
+        var colonIndex = line.IndexOf(':');
+       return line.Substring(colonIndex + 1).Trim();
+    }
+    throw new Exception("Failed to get default RID for system");
+}
+
+void DoArchive(string runtime, DirectoryPath inputFolder, FilePath outputFile)
+{
+    // On all platforms use ZIP for Windows runtimes
+    if (runtime.Contains("win"))
+    {
+        var zipFile = outputFile.AppendExtension("zip");
+        Zip(inputFolder, zipFile);
+    }
+    // On all platforms use TAR.GZ for Unix runtimes
+    else
+    {
+        var tarFile = outputFile.AppendExtension("tar.gz");
+        // Use 7z to create TAR.GZ on Windows
+        if (IsRunningOnWindows())
+        {
+            var tempFile = outputFile.AppendExtension("tar");
+            var exitCode = StartProcess("7z", new ProcessSettings
```

Comment:
Best way I found to do it is a try/catch block and a warning message if it did not find it. Default build is also Local so people should not trigger it, unless they explicitly want it.